example/lib/routes/biometrics_page.dart

```diff
@@ -0,0 +1,160 @@
+import 'package:flutter/material.dart';
+import 'package:gigya_flutter_plugin/gigya_flutter_plugin.dart';
+
+/// The Biometrics page.
+class BiometricsPage extends StatefulWidget {
+  /// Construct a [BiometricsPage] with the given [sdk].
+  const BiometricsPage({required this.sdk, super.key});
+
+  /// The [GigyaSdk] instance that is used by this widget.
+  final GigyaSdk sdk;
+
+  @override
+  State<BiometricsPage> createState() => _BiometricsPageState();
+}
+
+class _BiometricsPageState extends State<BiometricsPage> {
+  final GlobalKey<ScaffoldState> _scaffoldKey = GlobalKey<ScaffoldState>();
+
+  bool isOptIn = false;
+  bool isLocked = false;
+  String errorMessage = '';
+
+  void initializeBiometricState() async {
+    isLocked = await widget.sdk.biometricService.isLocked();
+    isOptIn = await widget.sdk.biometricService.isOptIn();
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    initializeBiometricState();
+  }
+
+  void _handleOptIn() async {
+    try {
+      await widget.sdk.biometricService.optIn(
+        parameters: <String, String>{
+          'title': 'SampleTitle',
+          'subtitle': 'SampleSubtitle',
+          'description': 'SampleDescription',
+        },
+      );
+    } catch (error) {
+      if (mounted) {
+        setState(() {
+          errorMessage = error.toString();
+        });
+      }
+    }
+  }
+
+  void _handleOptOut() async {
+    try {
+      await widget.sdk.biometricService.optOut(
+        parameters: <String, String>{
+          'title': 'SampleTitle',
+          'subtitle': 'SampleSubtitle',
+          'description': 'SampleDescription',
+        },
+      );
+    } catch (error) {
+      if (mounted) {
+        setState(() {
+          errorMessage = error.toString();
+        });
+      }
+    }
+  }
+
+  void _handleLockSession() async {
+    try {
+      await widget.sdk.biometricService.lockSession();
```

Comment:
Could you add this after the await? So that the state is refreshed.

```dart
isLocked = await widget.sdk.biometricService.isLocked();

if (mounted) {
  setState((){});
}
```